packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -804,6 +816,18 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   Color get _cursorColor => widget.cursorColor.withOpacity(_cursorBlinkOpacityController.value);
 
+  @override
+  bool get cutEnabled => !widget.readOnly;
```

Comment:
Maybe for a different PR, might be nice to let users explicitly control which of these are enabled or not.